src/helicopter.jl

```diff
@@ -0,0 +1,227 @@
+    Ωmr = 167 #rad/s
+    Rmr = 0.775 #m
+
+    ## DRAG PARAMETERS
+    # https://www.engineeringtoolbox.com/drag-coefficient-d_627.html
+    Cd = 1.17 #square flat plate 90degrees to the wind.
+
+    Sx_fus = 0.1 #m^2
+    Sy_fus = 0.22 #m^2
+    Sz_fus = 0.15 #m^2
+
+    ## TAIL ROTOR
+    ntr = 4.66 #Gear ratio of tr to mr
+    Ωtr = ntr * Ωmr #rad/s
+    Rtr = 0.13 #m
+    ltr = 0.91 #m
+    htr = 0.08 #m
+
+    ## AIR PARAMETERS
+    rho = 1.204 #kg/m^3 #ρ
+    
+
+    ####################################################################
+    #### X-CELL 60 SE Helicopter Dynamics: #############################
+    ####################################################################
+    ####################################################################
+    #=   
+    #### NOTE ON STATES::                                                   
+    x[14:16] --> wind velocities (estimated or ground-truth) 
+    x[14:16] --> wind "accelerations" 
+
+    #### NOTE ON DYNAMICS:
+    ASSUME: No blade-flapping dynamics;
+            No horizontal+Vertical fins;
+            No stabilizer;
+            Centre of Pressure == C.O.G
+
+    #### NOTE ON CONTROL INPUTS:
+    u[1] = Main Rotor tilt angle with vertical in Helicopter's X-Z plane (*check!) 
+    u[2] = Main Rotor tilt angle with vertical in Helicopter's Y-Z plane
+    u[3] = Main Rotor Thrust
+    u[4] = Tail Rotor Thrust
+    =#
+    ####################################################################
+    r = x[1:3] #x y z
+    q = x[4:7] # quaternion
+    v = x[8:10] #u v w
+    ω = x[11:13] #p q r
+    w = x[14:16] #wind-velocities
+    vw = x[17:19] #wind-velocity_dot
+
+    Q = qtoQ(q) #Rotation Matrix: [world_vector = Q * body_vector]
+    
+    ## MAIN ROTOR FORCES
+    Xmr = -u[3] * u[1] #Force along Body-X
+    Ymr = u[3] * u[2] #Force along Body-Y
+    Zmr = -u[3] #Force along Body-Z
+    
+    ## MAIN ROTOR MOMENTS
+    Lmr = Ymr * hmr #Moment about Body-X
+    Mmr = -Xmr * hmr #Moment about Body-Y
+
+    ## MAIN ROTOR YAW TORQUE AGAINST AIR
+    Qe = 0.0005 * rho * (Ωmr * Rmr)^2 * pi * Rmr^3  #(*Check Qe!)
+    
+    ## TAIL ROTOR FORCES
+    Ytr = u[4]
+    
+    ## TAIL ROTOR MOMENTS
+    Ntr = -Ytr * ltr #Moment about Body-Z
+    Ltr = Ytr * htr #Moment about Body-X
+    
+    ## DRAG FORCES
+    w_b = Q' * w #wind in Body Frame
+
+    # Note: need to take care of the sign manually...
+    Xfus = sign(w_b[1] - v[1]) * (0.5*rho*Sx_fus*Cd*(w_b[1] - v[1])*(w_b[1] - v[1])) #Drag force along Body-X
+    Yfus = sign(w_b[2] - v[2]) * (0.5*rho*Sy_fus*Cd*(w_b[2] - v[2])*(w_b[2] - v[2])) #Drag force along Body-Y
+    Zfus = sign(w_b[3] - v[3]) * (0.5*rho*Sz_fus*Cd*(w_b[3] - v[3])*(w_b[3] - v[3])) #Drag force along Body-Z
+
+    ## GRAVITY IN BODY-FRAME
+    g_matrix = Q' * [0; 0; -g]
+    
+    ## NEWTON-EULER'S EQUATIONS
+    #ṙ, q̇, v̇, ω̇ 
+    r_dot = Q * v
+    q_dot = 0.5 * L(q) * H * ω
+    v_dot = [(v[2] * ω[3]) - (v[3] * ω[2]) - g_matrix[1] + ((Xmr + Xfus) / M);

```

Comment:
Again, you can save an allocation by trivially making this a static array by placing an `SA` in front:
```
SA[
# my terms...
]
```